gulp-tasks/publish-bundle.js

```diff
@@ -0,0 +1,211 @@
+    tempPath,
+  ];
+
+  return spawnPromiseWrapper('git', params)
+  .then(() => tempPath);
+};
+
+const buildProject = (projectPath) => {
+  return spawnPromiseWrapper('npm', ['install'], {
+    cwd: projectPath,
+  })
+  .then(() => {
+    return spawnPromiseWrapper('gulp', ['build'], {
+      cwd: projectPath,
+    });
+  });
+};
+
+/**
+ * This function will create a directory with the same name at the
+ * .tar.gz file it generates. This way when the file is extracted
+ * the folder structure will have the sam
+ */
+const bundleProject = (projectPath, tagName) => {
+  const pattern = path.posix.join(projectPath, 'packages', '**',
+    constants.PACKAGE_BUILD_DIRNAME, constants.BROWSER_BUILD_DIRNAME,
+    '*.{js,map}');
+
+  const tarName = `workbox-${tagName}.tar.gz`;
+
+  const filesToIncludeInBundle = glob.sync(pattern);
+  const bundleDirectory = path.join(projectPath, '..', 'github-bundle');
+
+  filesToIncludeInBundle.forEach((fileToInclude) => {
+    fs.copySync(
+      fileToInclude,
+      path.join(bundleDirectory, path.basename(fileToInclude))
+    );
+  });
+
+  const tarball = new TarGz({}, {
+    // Do not inlcude the top level in the tar.gz
+    fromBase: true,
+  });
+  const tarPath = path.join(bundleDirectory, '..', tarName);
+  return tarball.compress(bundleDirectory, tarPath)
+  .then(() => {
+    return {
+      bundleDirectory,
+      tarPath,
+    };
+  });
+};
+
+const handleGithubAndCDNRelease = (tagName, gitBranch, release) => {
+  let releaseCreation = Promise.resolve();
+  if (!release) {
+    releaseCreation = github.repos.createRelease({
+      owner: GITHUB_OWNER,
+      repo: GITHUB_REPO,
+      tag_name: tagName,
+      draft: true,
+      name: tagName,
+    })
+    .then((response) => {
+      release = response.data;
+    });
+  }
+
+  return releaseCreation
+  .then(() => {
+    return downloadGitCommit(gitBranch);
+  })
+  .then((tagDownloadPath) => {
+    return buildProject(tagDownloadPath)
+    .then(() => {
+      return bundleProject(tagDownloadPath, tagName);
+    });
+  })
+  .then(({bundleDirectory, tarPath}) => {
+    return github.repos.uploadAsset({
+      owner: GITHUB_OWNER,
+      repo: GITHUB_REPO,
+      id: release.id,
+      filePath: tarPath,
+      name: path.basename(tarPath),
+      label: path.basename(tarPath),
+    })
+    .then(() => {
+      return uploadBundleToCDN(tagName, bundleDirectory);
+    });
+  });
+};
+
+const findTagsWithoutBundles = () => {
+  // Get all of the tags in the repo.
+  return github.repos.getTags({
+    owner: GITHUB_OWNER,
+    repo: GITHUB_REPO,
+  })
+  .then((tagsResponse) => {
+    // We only want tags that are v3.0.0 or above.
+    const tagsData = tagsResponse.data;
+    return tagsData.filter((tagData) => {
+      // The "-alpha" means semver will match against tags with prelease
+      // info (although the prelease tag must be a string >= 'alpha'
+      // alphabetically speaking).
+      // TODO: Change to v3.0.0 when v3 is launched.
```

Comment:
Can you put this in a CONSTANT at the top of the file for increased visibility as something that needs to change?